addons/mail/static/src/core/web/mail_core_web_service.js

```diff
@@ -38,7 +38,7 @@ export class MailCoreWeb {
                 inbox.counter++;
             }
             inbox.messages.add(message);
-            if (notifId > message.thread.message_needaction_counter_bus_id) {
+            if (notifId > message.thread?.message_needaction_counter_bus_id) {
```

Comment:
Ok because `integer > undefined` is falsy. IMO not intuitive so I'd prefer `message.thread &&`, just to avoid any clarity when reading these LOCs